src/System.Diagnostics.TraceSource/src/System/Diagnostics/SwitchAttribute.cs

```diff
@@ -48,43 +48,48 @@ public Type SwitchType
 
         public string SwitchDescription { get; set; }
 
+        private static IEnumerable<T> CastTo<T>(object[] source)
+        {
+            foreach (var item in source)
+                yield return (T)item;
+        }
```

Comment:
This adds extra overhead -- an enumerator allocation and it doesn't implement `ICollection<T>`, so `AddRange` can't go down the optimized path...

Maybe it'd be best to just use `List<object>` instead of `List<SwitchAttribute>`. That way you can pass the `object[]` to `AddRange`, while still removing the dependency on `ArrayList`.

Then, instead of doing `return switchAttribs.ToArray();` at the end, go back to:

```c#
SwitchAttribute[] ret = new SwitchAttribute[switchAttribs.Count];
switchAttribs.CopyTo(ret, 0);
return ret;
```